lib/utils.js

```diff
@@ -37,6 +37,40 @@ export function through (transform) {
   })
 }
 
+// DIY object pooling as a replacement for reusify
+export class ObjectPool {
+  constructor (ClassCtor) {
+    this.ClassCtor = ClassCtor
+    this.pool = []
+  }
+
+  get () {
+    return this.pool.pop() || new this.ClassCtor()
+  }
+
+  release (obj) {
+    // Optionally reset the object here if needed
+    this.pool.push(obj)
+  }
+}
+
+// Nodejs can clear timers in constant time these days
+// https://asafdav2.github.io/2017/node-js-timers/
+// Running retimers bench.js on Node 24.5 shows retimer to be 1 second faster on 1M invocations.
+// hence we can simplify retimer so we don't have to rely on a dependency
+export function retimer (fn, ms) {
+  let timeout = setTimeout(fn, ms)
+  return {
+    reschedule (newMs) {
+      clearTimeout(timeout)
+      timeout = setTimeout(fn, newMs)
+    },
+    clear () {
+      clearTimeout(timeout)
+    }
+  }
+}
```

Comment:
Do this instead https://nodejs.org/docs/latest/api/timers.html#timeoutrefresh.

Reply:
applied